Author: Michael Piefel <piefel@debian.org>
Description: FIXME, and #417511.

--- a/doc/file.man
+++ b/doc/file.man
@@ -56,7 +56,7 @@
 or non-printable).
 Exceptions are well-known file formats (core files, tar archives)
 that are known to contain binary data.
-When modifying magic files or the program itself, make sure to
+When adding local definitions to /etc/magic, make sure to
 .Em "preserve these keywords" .
 Users depend on knowing that all the readable files in a directory
 have the word
@@ -100,7 +100,8 @@
 has been applied by extension to data files.
 Any file with some invariant identifier at a small fixed
 offset into the file can usually be described in this way.
-The information identifying these files is read from the compiled
+The information identifying these files is read from /etc/magic
+and the compiled
 magic file
 .Pa __MAGIC__.mgc ,
 or the files in the directory
@@ -453,12 +454,6 @@
 The order of entries in the magic file is significant.
 Depending on what system you are using, the order that
 they are put together may be incorrect.
-If your old
-.Nm
-command uses a magic file,
-keep the old magic file around for comparison purposes
-(rename it to
-.Pa __MAGIC__.orig ) .
 .Sh EXAMPLES
 .Bd -literal -offset indent
 $ file file.c file /dev/{wd0a,hda}
