Author: Daniel Baumann <mail@daniel-baumann.ch>
Description: Updating gzip mime (Closes: #688886).

--- a/magic/Magdir/compress
+++ b/magic/Magdir/compress
@@ -21,7 +21,7 @@
 #	* Produce shorter output - notably, only report compression methods
 #         other than 8 ("deflate", the only method defined in RFC 1952).
 0       string          \037\213        gzip compressed data
-!:mime	application/x-gzip
+!:mime	application/gzip
 !:strength * 2
 >2	byte		<8		\b, reserved method
 >2	byte		>8		\b, unknown method
