Author: chrysn <chrysn@fsfe.org>
Description: Adding new magics for LXT (Closes: #647412).

diff -Naurp file.orig/magic/Magdir/scientific file/magic/Magdir/scientific
--- file.orig/magic/Magdir/scientific
+++ file/magic/Magdir/scientific
@@ -104,3 +104,8 @@
 >>5	byte	x		version %d.0
 >4	byte	>0x00		version %d
 >>5	byte	x		\b.%d
+
+# Type: LXT (interLaced eXtensible Trace)
+# chrysn <chrysn@fsfe.org>
+0	beshort	0x0138	interLaced eXtensible Trace (LXT) file
+>2	beshort	>0	(Version %u)
