Subject: --parameter takes a parameter
Origin: FILE5_24-22-g27b4e34
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Tue Sep 8 13:46:01 2015 +0000

--- a/src/file_opts.h
+++ b/src/file_opts.h
@@ -43,7 +43,7 @@
 #if defined(HAVE_UTIME) || defined(HAVE_UTIMES)
 OPT('p', "preserve-date", 0, "        preserve access times on files\n")
 #endif
-OPT('P', "parameter", 0, "            set file engine parameter limits\n"
+OPT('P', "parameter", 1, "            set file engine parameter limits\n"
     "                               indir        15 recursion limit for indirection\n"
     "                               name         30 use limit for name/use magic\n"
     "                               elf_notes   256 max ELF notes processed\n"
