Subject: apply parameters before unwrapping() the list of files
Origin: FILE5_24-23-g4ddb783
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Tue Sep 8 13:46:49 2015 +0000

--- a/src/file.c
+++ b/src/file.c
@@ -233,6 +233,7 @@
 			if (magic == NULL)
 				if ((magic = load(magicfile, flags)) == NULL)
 					return 1;
+			applyparam(magic);
 			e |= unwrap(magic, optarg);
 			++didsomefiles;
 			break;
